test/spec-wasm2c-prefix.c

```diff
@@ -342,31 +342,31 @@ static w2c_spectest spectest_instance;
 /*
  * spectest implementations
  */
-void w2c_spectest_print(w2c_spectest* instance) {
+void w2c_spectest_print(void* instance) {
   printf("spectest.print()\n");
 }
 
-void w2c_spectest_print_i32(w2c_spectest* instance, uint32_t i) {
+void w2c_spectest_print_i32(void* instance, uint32_t i) {
   printf("spectest.print_i32(%d)\n", i);
 }
 
-void w2c_spectest_print_i64(w2c_spectest* instance, uint64_t i) {
+void w2c_spectest_print_i64(void* instance, uint64_t i) {
   printf("spectest.print_i64(%" PRIu64 ")\n", i);
 }
 
-void w2c_spectest_print_f32(w2c_spectest* instance, float f) {
+void w2c_spectest_print_f32(void* instance, float f) {
   printf("spectest.print_f32(%g)\n", f);
 }
 
-void w2c_spectest_print_i32_f32(w2c_spectest* instance, uint32_t i, float f) {
+void w2c_spectest_print_i32_f32(void* instance, uint32_t i, float f) {
   printf("spectest.print_i32_f32(%d %g)\n", i, f);
 }
 
-void w2c_spectest_print_f64(w2c_spectest* instance, double d) {
+void w2c_spectest_print_f64(void* instance, double d) {
   printf("spectest.print_f64(%g)\n", d);
 }
 
-void w2c_spectest_print_f64_f64(w2c_spectest* instance, double d1, double d2) {
+void w2c_spectest_print_f64_f64(void* instance, double d1, double d2) {
```

Comment:
Using `void*` everwhere seems strictly worse.  Could you explain (perhaps in the PR description) why this is needed?

Reply:
It's not *strictly* needed, the alternative is to emit wrapper functions for anything that goes into a table, tho this way was *significantly* less effort.

Reply:
done(?)